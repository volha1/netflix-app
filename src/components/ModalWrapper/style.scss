@import '../../common/scss/base.scss';

.modal {
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: none;
  background: rgba(23, 23, 23, 0.5);
  z-index: 10;
}

.modal-content {
  padding: 25px;
  background: $black;
}

.modal.active {
  display: flex;
  justify-content: center;
  align-items: center;
}
