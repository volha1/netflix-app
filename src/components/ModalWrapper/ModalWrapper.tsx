import React, { ReactElement } from 'react';
import './style.scss';

type ChangeMovieWindowProps = {
  children: ReactElement;
  visible: boolean;
};

const ModalWrapper = ({ children, visible }: ChangeMovieWindowProps): ReactElement => {
  const classes = ['modal'];

  if (visible) {
    classes.push('active');
  }

  return (
    <div className={classes.join(' ')}>
      <div
        className="modal-content"
        onClick={(e): void => {
          return e.stopPropagation();
        }}
      >
        {children}
      </div>
    </div>
  );
};

export default ModalWrapper;
