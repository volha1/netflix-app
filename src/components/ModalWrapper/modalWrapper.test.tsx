import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import ModalWrapper from './ModalWrapper';

describe('ModalWrapper test', () => {
  it('ModalWrapper snapshot', () => {
    const component = renderer.create(
      <ModalWrapper isVisible>
        <p>test</p>
      </ModalWrapper>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('ModalWrapper children should render', () => {
    render(
      <ModalWrapper isVisible>
        <div data-testid="test-data">test</div>
      </ModalWrapper>
    );

    expect(screen.getByTestId('test-data')).toBeInTheDocument();
  });
});
