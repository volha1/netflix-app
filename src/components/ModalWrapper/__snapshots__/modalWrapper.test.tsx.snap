// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper test ModalWrapper snapshot 1`] = `
<div
  className="modal active"
>
  <div
    className="modal-content"
    onClick={[Function]}
  >
    <p>
      test
    </p>
  </div>
</div>
`;
